internal/bminventory/inventory.go

```diff
@@ -1160,15 +1160,15 @@ func (b *bareMetalInventory) GenerateInfraEnvISOInternal(ctx context.Context, in
 	}
 
 	if imageExists {
-		if err = b.updateImageInfoPostUpload(ctx, infraEnv, infraEnv.ProxyHash, infraEnv.Type, false); err != nil {
```

Comment:
I think that we can set the image type into the InfraEnv at the same time we set all the rest of the parameters (before calling GenerateClusterISOInternal). That way it will be more consistent with the update once we move to image service. And we have Generated field that speicifies if image was created or not

Reply:
I was trying to be consistent with the v1 generate here where we only update the image type after it's created successfully.

If we fail to create, then we'll end up with the wrong image type, right?

Reply:
Plus we have to pass something here.

Reply:
before creating image we set the "generated field to false, and updated it to true only after generation has succeeded. So even if we set the image type before creation, and creation fails, then next time GenerateclusterISO is called we check the "generated" and since it false we try to create the image again. So eventually the mage is created with the right type ( i think  :)).